backend/src/test/java/com/google/sps/UserDatastoreTest.java

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.User;
+import com.google.sps.datastore.UserDatastore;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class UserDatastoreTest {
+
+  // Default values
+  private static final long ID_DEFAULT = 0;
+
+  // Some usernames
+  private static final String PERSON_A = "Person A";
+  private static final String PERSON_B = "Person B";
+
+  // Datastore Key/Property constants
+  private static final String KIND_USER = "User";
+  private static final String PROPERTY_USERNAME = "username";
+
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
```

Comment:
Out of scope for this PR but it would be nice to either find a corresponding JUnit Rule for this or just write our own. A JUnit Rule encapsulates some setUp and tearDown work so that it can be reused across tests without repeating the same code multiple times. 

Since you'll no doubt use JUnit in the future if you continue working with Java I'd recommend filing an issue and investigating.